roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md

```diff
@@ -65,82 +78,75 @@ Ephemeral in-memory state:
 
 ```rust
 struct State {
-    keystore: KeyStore,
-    highest_session: SessionIndex,
+  keystore: Arc<LocalKeystore>,
+	rolling_session_window: RollingSessionWindow,
+	highest_session: SessionIndex,
+	spam_slots: SpamSlots,
+	participation: Participation,
+	ordering_provider: OrderingProvider,
+	participation_receiver: WorkerMessageReceiver,
+	metrics: Metrics,
+	// This tracks only rolling session window failures.
+	// It can be a `Vec` if the need to track more arises.
+	error: Option<SessionsUnavailable>,
+	/// Latest relay blocks that have been successfully scraped.
+	last_scraped_blocks: LruCache<Hash, ()>,
 }
 ```
 
 ### On startup
+Wait for first leaf from Runtime.
```

Comment:
Leaves don't come from runtime, but the overseer - which gets them on chain import - no runtime calls are involved here.